sys/net/application_layer/coapfileserver/doc.txt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2020 chrysn
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_coapfileserver CoAP file server
+ * @ingroup     net
+ * @brief       Library for serving files from the VFS to CoAP clients
+ *
+ * # About
+ * 
+ * This maps files in the local file system onto a resources in CoAP. In that,
+ * is is what is called a static web server in the unconstrained web.
+ *
+ * As usual, GET operations are used to read files<!-- WRITESUPPORT, and PUT writes to files.
+ * In the current implementation, PUTs are expressed as random-access, meaning
+ * that files are not updated atomically -->.
```

Comment:
Because I had write support planned to go in from the start, had the documentation already written in the first step, and don't want to have to write it all again later on. It'll better if I move them over to a nascent write-support-for-fileserver branch instead.